### PR TITLE
Implement template collections

### PR DESCRIPTION
**Pull request by Danappelxx**
Hopefully my usage of Foundation works on linux 😬

### CODE REVIEW

File: Sources/Template.swift [hunk 1]
```diff
@@ -68,3 +68,8 @@ public struct Template {
         return Renderer(ast: ast, partials: partials.mapValues { $0.ast }).render(with: context)
     }
 }
+
+extension Template: Equatable {}
+public func ==(lhs: Template, rhs: Template) -> Bool {
```
Comment (paulofaria) [inline Sources/Template.swift]: this now can be implemented as `public static func` inside the extension. just so you know! (:

File: Sources/TemplateCollection.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
+public enum TemplateCollectionError: Error {
+    case noSuchTemplate(named: String)
+}
+
+public struct TemplateCollection {
+    public var templates: [String:Template]
```
Comment (paulofaria) [inline Sources/TemplateCollection.swift]: nit: `[String: Template]`

File: Sources/TemplateCollection.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+// MARK: IO
+import Foundation
+extension TemplateCollection {
+    public init(basePath: String = FileManager.default.currentDirectoryPath, directory: String) throws {
+        let path = basePath + directory
+        let files = try FileManager.default.contentsOfDirectory(atPath: path)
+
+        var templates = [String:Template]()
+
+        for file in files where file.split(separator: ".").last == "mustache" {
```
Comment (paulofaria) [inline Sources/TemplateCollection.swift]: maybe make the extension configurable?

Comment (paulofaria) [inline Sources/TemplateCollection.swift]: also.. `Foundation` has a property to get the file extension.. I think it's on `NSString` IIRC

File: Sources/TemplateCollection.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            guard
+                let handle = FileHandle(forReadingAtPath: "\(path)/\(file)"),
+                let contents = String(data: handle.readDataToEndOfFile(), encoding: .utf8)
+                else {
+                continue
+            }
+
+            let template = try Template(contents)
+            // remove .mustache extension
+            let name = file.split(separator: ".").dropLast().joined(separator: ".")
```
Comment (paulofaria) [inline Sources/TemplateCollection.swift]: same here. `Foundation` has a method for this on `NSString` IIRC

File: Sources/TemplateCollection.swift (hunk 3 above)
Comment (Danappelxx) [inline Sources/TemplateCollection.swift]: What do you mean by configurable? Allow injection of file managers?

Comment (Danappelxx) [inline Sources/TemplateCollection.swift]: Oh, you meant file extensions haha. Yeah, will do.

**Pull request merged**